tests/catalog/test_rest.py

```diff
@@ -1646,6 +1646,40 @@ def test_rest_catalog_with_unsupported_auth_type() -> None:
     assert "Could not load AuthManager class for 'unsupported'" in str(e.value)
 
 
+def test_rest_catalog_with_oauth2_auth_type(requests_mock: Mocker) -> None:
```

Comment:
It would be nice to have a test where fetching the token does not return a 200 or where refreshing the token does not return a 200. 

That would allow us to verify that the non-happy paths work as intended.

Reply:
Done!